contributing.md

```diff
@@ -1,32 +1,196 @@
+
+# Overview
+
+A PR is simply a set of changes that the author would like to be added to a repository. In the EuroPi repository, the most common example would be a new script in the contrib directory. The PR process allows the maintainers of the repository manage the contributions in a consistent way, and provides a means for non-maintainers to add changes to the repository.
+
+The PR process consists of a few steps. First, the author makes their desired changes on a branch in their own fork of the main EuroPi repository. Second, the author opens a PR, submitting the changes for review. Next, the reviewers and author work together to ensure that the changes are appropriate for the repository. Finally, the changes are merged into the main repository. The details of this process are described below.
+
+If you need any help with any stage of this process (or anything else), please don't hesitate to ask questions in the [Discord Server](https://discord.gg/2eFyqP2rSs), [Discussions page](https://github.com/Allen-Synthesis/EuroPi/discussions), or email Rory directly at [contact@allensynthesis.co.uk](mailto:contact@allensynthesis.co.uk).
+
+
+# Responsibilities of PR Participants
+
+It is expected that everyone participating in the PR process acts in a friendly, supportive, and respectful manner. Always assume that teh other participants are acting in good faith, towards the common goal of improving the EuroPi project.
+
+## Responsibilities of the PR author
+
+It is the responsibility of the PR author to follow the [guidelines](#pr-guidelines-and-requirements) outlined in this document. Additionally, the PR author should, to the best of their ability, provide a complete and well thought out PR that is ready to merge.
+
+## Responsibilities of the PR reviewer
+
+It is the responsibility of the PR reviewer(s) to guide the author through the review process. They should verify that the author understands their responsibilities as outlined here and what they need to do at each step of the process.
+
+When commenting on a PR, the reviewer should make the intention of each comment clear. Specifically, is the reviewer requesting a change, and is that change required for PR approval. This is done by prefixing each comment with one of the following labels **\[required\]**, **\[optional\]**, **\[question\]**, **\[discussion\]**. A more detailed discussion of these labels can be found [below](#comment-labels).
+
+It is the responsibility of the PR reviewer(s) to reasonably understand the change that they are reviewing, as they are responsible for ensuring that the proposed change has a positive impact on the project and aligns with the project's goals. On the surface, this is done by enforcing the PR guidelines and steering the author towards best practices. At a deeper level, the reviewer must also consider the impact of the PR in the scope of the larger EuroPi project. This is particularly important on changes that impact the EuroPi libraries or documentation.
```

Comment:
> It is the responsibility of the PR reviewer(s) to reasonably understand the change that they are reviewing, as they are responsible for ensuring that the proposed change has a positive impact on the project and aligns with the project's goals.

Additionally, I think it may be a good idea if the reviewer reserves the right to reject / close a PR that would require an excessive amount of feedback, for example an inexperienced coder requesting review of a script with many egregious best-practice violations.

Reply:
I'm not sure how I feel about this. I think I'd be ok with helping someone improve a PR like this, assuming that the idea showed some promise and they were willing to put in the work and understood the time that it would involve.

I think that the only time I'd personally want to close someone else's PR is if the consensus of the reviewers was that the changes were not appropriate for the project.

Reply:
That sounds good to me. Perhaps my suggestion should be more viewed as a edge case rather than needing to be a documented rule.